tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -53,18 +54,26 @@ get_interval_from_config(const Dimension *dim, const Jsonb *config, const char *
 	*isnull = false;
 	if (IS_INTEGER_TYPE(partitioning_type))
 	{
-		bool found;
+		bool found, overflow;
+		int64 res = 0;
 		int64 interval_val = ts_jsonb_get_int64_field(config, json_label, &found);
 		if (!found)
 		{
 			*isnull = true;
-			return (Datum) 0;
+			return 0;
 		}
 		Oid now_func = ts_get_integer_now_func(dim);
 
 		Assert(now_func);
 
-		return subtract_integer_from_now(interval_val, partitioning_type, now_func);
+		res = subtract_integer_from_now_internal(interval_val,
+												 partitioning_type,
+												 now_func,
+												 &overflow);
+		/* overflowed, so return max for that data type */
+		if (overflow)
+			res = ts_time_get_max(partitioning_type);
```

Comment:
If we are doing subtraction, shouldn't we return min instead of max? I guess the interval could be negative, in which case the subtraction effectively becomes an addition. It seems we need to check for overflow in each end of the valid time range.

Let's say you have a smallint table and do this:

```
SELECT add_continuous_aggregate_policy('cagg', 50000, -50000, '1 hour');
```

This should basically be turned into a refresh of [MIN, MAX], but, if I am not mistaken, this will become [ MAX, MAX] with the current code.